lua/notagain/essential/autorun/server/realistic_falldamage.lua

```diff
@@ -14,44 +19,37 @@ hook.Add("Move", "realistic_falldamage", function(ply, data)
 			local res = util.TraceHull(params)
 
 			local dmg = (len - 500) / 4
+			local z = math.abs(res.HitNormal.z)
 
-			if not res.HitWorld and res.Entity:IsValid() then
-				local owner = res.Entity.CPPIGetOwner and res.Entity:CPPIGetOwner()
-				if owner and not owner:CanAlter(ply) then return end
+			if res.Hit and (z < 0.1 or z > 0.9) then
+				local fall_damage = GetFallDamage(ply, len) -- Prepare Override & Check Expected Fall Damage
+				if fall_damage <= 0 then return end
 
+				local pos = data:GetOrigin()
 				local info = DamageInfo()
-				info:SetDamagePosition(data:GetOrigin())
+				info:SetDamagePosition(pos)
 				info:SetDamage(dmg)
 				info:SetDamageType(DMG_FALL)
-				info:SetAttacker(ply)
-				info:SetInflictor(ply)
+				info:SetAttacker(Entity(0))
+				info:SetInflictor(Entity(0))
 				info:SetDamageForce(ply.fdmg_last_vel)
-				res.Entity:TakeDamageInfo(info)
-			end
-
-			local z = math.abs(res.HitNormal.z)
 
-			if res.Hit and (z < 0.1 or z > 0.9) then
-				local fall_damage = hook.Run("GetFallDamage", ply, len)
-				if fall_damage ~= 0 then
-					if fall_damage < dmg then
-						dmg = dmg - fall_damage
-						local pos = data:GetOrigin()
-						if hook.Run("RealisticFallDamage", ply, pos, dmg, len, res, params) ~= true then
-							local info = DamageInfo()
-							info:SetDamagePosition(pos)
-							info:SetDamage(dmg)
-							info:SetDamageType(DMG_FALL)
-							info:SetAttacker(Entity(0))
-							info:SetInflictor(Entity(0))
-							info:SetDamageForce(ply.fdmg_last_vel)
-							ply:TakeDamageInfo(info)
-						end
-					end
+				if hook.Run("RealisticFallDamage", ply, info, len, dmg, fall_damage, res, params) ~= true then
+					ply:TakeDamageInfo(info)
 				end
 			end
 		end
 	end
 
 	ply.fdmg_last_vel = vel
-end)
\ No newline at end of file
+end)
+
```

Comment:
I would suggest using a local variable inside the script to suppress damage instead of adding it to the player table. This makes the hack local to this file.
```
local suppress_fall_damage
...
suppress_fall_damage = true
ply:TakeDamageInfo(info)
suppress_fall_damage = false
...
hook.Add("EntityTakeDamage", "realistic_falldamage", function(ply, di)
    if suppress_fall_damage then return true end
end)
```